File: python/questions/jump.py
from typing import List


class Solution:
    def jump(self, nums: List[int]) -> int:
        ln = len(nums)
        l = 0
        r = 0
        s = 0
        while l <= r:
            if r >= ln - 1:
                return s
            cr = r
            for p in range(l, cr + 1):
                r = max(r, nums[p] + p)
            l = cr + 1
            s += 1
        return -1

File: typescript/questions/jump.ts
function jump(nums: number[]): number {
  let nl = nums.length;
  let l = 0;
  let r = 0;
  let s = 0;
  while (l <= r) {
    if (r >= nl - 1) {
      return s;
    }
    let cr = r;
    for (; l <= cr; ++l) {
      r = Math.max(r, nums[l] + l);
    }
    ++s;
  }
  return -1;
}
